boolean/boolean.py

```diff
@@ -1003,14 +1009,70 @@ def symbols(*args):
     "(": 20,
 }
 
+ANDS = ('*', '&', 'and')
+ORS = ('+', '|', 'or')
+NOTS = ('~', '!', 'not')
+PARENS = ('(', ')')
+OPS = ANDS + ORS + NOTS + PARENS
+ZERO_OR_ONE = ('0', '1')
+
 
-def parse(expr, eval=True):
+def tokenizer(expr):
     """
-    Returns a boolean expression created from the given string.
+    Return an iterable of tokens as tuples of (is_symbol, token, start row,
+    start column) given an `expr`.
+    `is_symbol` must be True if the token is a symbol. 
+    
+    If `is_symbol` the token must of an operation token or and opening '(' or
+    closing parenthesis ')'.
+
+    Recognized operations tokens are:
+    - for and:  '*', '&', 'and'
+    - for or: '+', '|', 'or'
+    - for not: '~', '!', 'not'
     """
     if not isinstance(expr, basestring):
         raise TypeError("Argument must be string but it is %s." % type(expr))
 
+    ignored_token_types = (
+        tokenize.NL, tokenize.NEWLINE, tokenize.COMMENT,
+        tokenize.INDENT, tokenize.DEDENT,
+        tokenize.ENDMARKER
+    )
+
+    OPS = ANDS + ORS + NOTS + PARENS
+
+    # note: unbalanced expression may raise a TokenError here.
+    tokens = tokenize.generate_tokens(StringIO(expr).readline)
+
+    for toktype, tok, (srow, scol,), _, _ in tokens:
+        if toktype in ignored_token_types or not tok.strip():
+            continue
+
+        is_symbol = False
+        if tok.lower() in OPS:
+            is_symbol = False
+            tok = tok.lower()
+        elif toktype == tokenize.NAME or tok in ZERO_OR_ONE:
+            is_symbol = True
+        else:
+            raise TypeError("Unknown token: %(tok)s at line: %(srow)d, column: %(scol)d" % locals())
+
+        yield is_symbol, tok, srow, scol
```

Comment:
`some:name` or `some.name` are no valid python identifiers, though in the `some.name` each part is.
My approach would be to only support by default valid identifiers, and then if you want to support other names, you can create your own `tokenizer` function such as in this example: https://github.com/bastikr/boolean.py/pull/23/files/1908bc01fc48f754f140b2ecab2b12e6806f8a26#diff-63c8b2556011726f89a5fe31e9c596f0R72